src/appleseed/renderer/modeling/aov/cryptomatte.cpp

```diff
@@ -0,0 +1,908 @@
+        {
+            uint32 m3hash = 0;
+            string obj_name;
+            if (shading_point.hit_surface())
+            {
+                switch (m_layer_type)
+                {
+                  case CryptomatteAOV::CryptomatteType::ObjectNames:
+                    obj_name = shading_point.get_object().get_name();
+                    break;
+
+                  case CryptomatteAOV::CryptomatteType::MaterialNames:
+                    {
+                      const auto* obj_material = shading_point.get_material();
+                      if (obj_material != nullptr)
+                          obj_name = obj_material->get_name();
+                    }
+                    break;
+
+                  default:
+                    break;
+                }
+
+                MurmurHash3_x86_32(reinterpret_cast<const unsigned char*>(obj_name.c_str()), static_cast<int>(obj_name.size()), 0, &m3hash);
+            }
+
+            const Vector2i pixel_pos = pixel_context.get_pixel_coords();
+            const Vector2d sample_pos = pixel_context.get_sample_position();
+            int tile_pixel_x = pixel_pos.x - m_tile_origin_x;
+            int tile_pixel_y = pixel_pos.y - m_tile_origin_y;
+
+            m_tile_name_map[make_pair(m_tile_origin_x, m_tile_origin_y)][m3hash] = obj_name;    //tile_origin indexing of the map should help avoiding the race condition
+
+            double sample_pos_x = (sample_pos.x / m_rcp_canvas_width) - pixel_pos.x;
+            double sample_pos_y = (sample_pos.y / m_rcp_canvas_height) - pixel_pos.y;
+
+            AABB2u tile_crop_window(
+                Vector2u(m_crop_window.min.x - m_tile_origin_x, m_crop_window.min.y - m_tile_origin_y),
+                Vector2u(m_crop_window.max.x - m_tile_origin_x, m_crop_window.max.y - m_tile_origin_y)
+            );
+            
+            filter(
+                static_cast<float>(tile_pixel_x + sample_pos_x),
+                static_cast<float>(tile_pixel_y + sample_pos_y),
+                tile_crop_window,
+                m3hash);
+        }
+
+    private:
+        const Filter2f*                 m_renderer_filter;
+        int                             m_tile_origin_x;
+        int                             m_tile_origin_y;
+        int                             m_tile_end_x;
+        int                             m_tile_end_y;
+        size_t                          m_frame_width;
+        double                          m_rcp_canvas_width;
+        double                          m_rcp_canvas_height;
+        AABB2u                          m_crop_window;
+        Image&                          m_aov_image;
+        size_t                          m_num_layers;
+        vector<WeightMap>&              m_pixel_samples;
+        map<pair<int, int>, NameMap>&   m_tile_name_map;
+        CryptomatteAOV::CryptomatteType m_layer_type;
+
+        // Copy of the FilteredTile::add().
```

Comment:
Remove "the" or add "method" at the end.

Maybe "inspired" instead of "copy" since it's not actually a copy?